src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -435,9 +442,24 @@ Status GcsActorManager::RegisterActor(
   // The backend storage is supposed to be reliable, so the status must be ok.
   RAY_CHECK_OK(gcs_table_storage_->ActorTable().Put(
       actor->GetActorID(), *actor->GetMutableActorTableData(),
-      [this, actor](const Status &status) {
+      [this, actor, actor_already_destroyed_callback](const Status &status) {
         // The backend storage is supposed to be reliable, so the status must be ok.
         RAY_CHECK_OK(status);
+        // If an owner dies before this callback is called, the actor could have been
+        // already destroyed. In this case, we just mark this actor as dead.
+        auto registered_actor_it = registered_actors_.find(actor->GetActorID());
+        if (registered_actor_it == registered_actors_.end()) {
+          auto mutable_actor_table_data = actor->GetMutableActorTableData();
+          mutable_actor_table_data->set_state(rpc::ActorTableData::DEAD);
+          auto actor_table_data =
+              std::make_shared<rpc::ActorTableData>(*mutable_actor_table_data);
+          // Make sure to flush the actor state as DEAD to avoid overwriting it.
+          RAY_CHECK_OK(gcs_table_storage_->ActorTable().Put(actor->GetActorID(),
+                                                            *actor_table_data, nullptr));
```

Comment:
Why do not use *mutable_actor_table_data instead of *actor_table_data?

Reply:
Ah good point.. I just copied from `DestroyActor`. I think it is to avoid race condition (although we won't have any race condition in this case because it is single threaded). I will just flush `mutable_actor_table_data `